js/editor-ta.js

```diff
@@ -0,0 +1,304 @@
+/**
+ * @constructor
+ * @param {DOM} elementId
+ * @param {Settings} settings
+ */
+function EditorTextArea(editorElement, settings) {
+  this.element_ = editorElement;
+  this.settings_ = settings;
+  this.numLines_ = 1;
+
+  this.setTheme();
+
+  const initFontSize = this.settings_.get('fontsize') + 'px';
+
+  // The main text area the user will be interacting with
+  this.textarea_ = document.createElement('textarea');
+  this.textarea_.setAttribute('id', 'editor-ta');
+  this.textarea_.style.fontSize = initFontSize;
+  this.textarea_.addEventListener('input', function() {
+    this.onChange();
+  }.bind(this));
+  window.addEventListener('resize', function() {
+    this.onResize();
+  }.bind(this));
+
+  this.wrapper_ = document.createElement('div');
+  this.wrapper_.classList.add('editor-ta-wrapper');
+  this.wrapper_.appendChild(this.textarea_);
```

Comment:
I'd like to keep it as is since codemirror does not have any UI and i really want to unify the interface that both codemirror and the text area use to minimize the chance of things going weird when you switch between the two.

Reply:
Ack. Maybe add a comment as such?

Reply:
Will do!